app/services/stripe_service/api/payments.rb

```diff
@@ -48,8 +62,25 @@ def create_preauth_payment(tx, gateway_fields)
           buyer_commission_cents: buyer_commission.cents,
           fee_cents: fee.cents,
           subtotal_cents: subtotal.cents,
-          stripe_charge_id: stripe_charge.id
-        })
+          stripe_charge_id: stripe_charge.try(:id)
+        }
+        if intent
+          payment_data[:stripe_payment_intent_id] = intent.id
+          if intent.status == 'requires_action' &&
+             intent.next_action.type == 'use_stripe_sdk'
```

Comment:
I think we shouldn't only handle `use_stripe_sdk`. It's possible to get a redirect action too, I think, but Stripe.js should be able to handle that in the UI too via iframe. Was there a reason whey next_action.type is checked here?

Reply:
@bladealslayer 
No way to check something in the other place. This wrapped into the large count of methods. Most safe communication with front-end is via DB as this was implemented. Stripe docs advise to use this code. I could not reply about some Stripe internals.

Reply:
Accordig to Stripe docs another `next_action` type could be `redirect_to_url`, but to be honest I don't know under what conditions that can happen - when is it going to be `use_stripe_sdk` and when `redirect_to_url`.

https://stripe.com/docs/api/payment_intents/object#payment_intent_object-next_action

Can you show me which example you followed?

Reply:
Example https://stripe.com/docs/payments/payment-intents/quickstart#creating-with-manual-confirmation 
Yes, the current implementation does not take into account redirect_to_url. Do you have any idea how to trigger this case? Our case is confirming a PaymentIntent with Stripe.js. As far I get our case does not trigger redirect_to_url in the Stripe response.

Reply:
I've contacted Stripe support about this to get a clarification.

Reply:
This is ok, type will always be `use_stripe_sdk` with this type of PaymentIntent handling.